### PR TITLE
Setup CI

### PR DESCRIPTION
**Pull request by obrok**
Setup a dockerized CI build. Not touching the `test.sh` for now - I'm planning to work on that tomorrow.

### CODE REVIEW

File: test-project/src/test_contract.rs [hunk 1]
```diff
@@ -53,8 +53,8 @@ impl Instance {
             data
         };
         let code_hash = [
-            41, 2, 122, 125, 224, 108, 110, 209, 243, 246, 117, 145, 214, 14, 89, 13, 162, 122,
-            197, 124, 57, 21, 90, 127, 101, 94, 152, 68, 215, 159, 216, 127,
+            175, 190, 2, 39, 246, 101, 99, 93, 122, 83, 182, 185, 58, 200, 206, 0, 165, 35, 80,
```
Comment (deuszx) [inline test-project/src/test_contract.rs:56]: Why did we have to update code hashes if nothing about the contracts changed?

File: ci/Dockerfile.aleph-node [hunk 2]
```diff
@@ -0,0 +1,26 @@
+FROM public.ecr.aws/p6e8q1z1/aleph-node:r-10.0
+
+# Save node and sudo account ids for convenience
+RUN aleph-node key inspect //Alice | grep "SS58 Address" | awk "{print \$3}" \
+  > /node/sudo_account_id
+RUN aleph-node key inspect //1 | grep "SS58 Address" | awk "{print \$3}" \
+  > /node/node_account_id
```
Comment (deuszx) [inline ci/Dockerfile.aleph-node:7]: IIUC, the advise for writing leaner docker containers is to batch commands together if they are not expected to be updated independently:
```suggestion
RUN aleph-node key inspect //Alice | grep "SS58 Address" | awk "{print \$3}" > /node/sudo_account_id \
      && aleph-node key inspect //1 | grep "SS58 Address" | awk "{print \$3}" > /node/node_account_id
```

Comment (deuszx) [reply ci/Dockerfile.aleph-node:7]: The things below could probably be also included in that single RUN command. Unless you think you might need to change them w/o changing key generation (which I don't think is the case).

Fewer layers == smaller docker containers.

File: README.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
+# ink-wrapper
+
+## CI
+
+You should be able to run the CI processes locally with https://github.com/nektos/act:
```
Comment (deuszx) [inline README.md:5]: Personally, I like to create a Makefile with the commands required to build docker, start it and run the test. They can be packaged together into `make run-ci` and called from the CI as well. That way there would be a single point where things can change (Makefile) and they don't require installation of external tool (like this `act` here).

**Review by deuszx — APPROVED** [review]
No blockers but I left some comments.

File: test-project/src/test_contract.rs (hunk 1 above)
Comment (obrok) [reply test-project/src/test_contract.rs:56]: I don't know, I'm not an ink compiler :)

File: ci/Dockerfile.aleph-node (hunk 2 above)
Comment (obrok) [reply ci/Dockerfile.aleph-node:7]: :+1: will tweak that

File: test-project/src/test_contract.rs (hunk 1 above)
Comment (deuszx) [reply test-project/src/test_contract.rs:56]: Fair, I can accept that explanation.

File: README.md (hunk 3 above)
Comment (obrok) [reply README.md:5]: Let me do that in the next PR - I'll just cut up the `test.sh` file into pieces in a `Makefile`. I think this is slightly different from running the whole CI with `act` - the CI will still do `docker run ... make run-ci`, but locally you might want to run against a node run on the host, etc.

Comment (deuszx) [reply README.md:5]: OK.

**Pull request merged**